scripts/Install-AllSoftware.ps1

```diff
@@ -0,0 +1,24 @@
+# 1. Make sure the Microsoft App Installer is installed:
+#    https://www.microsoft.com/en-us/p/app-installer/9nblggh4nns1
+# 2. Run this script as administrator. If running scripts is blocked, you can temporarily unblock them by running 
+	Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process
+
+Write-Output "Installing apps"
+$apps = @(
+	@{name = "GitHub.GitLFS" },
+	@{name = "icsharpcode.ILSpy" },
+	@{name = "Microsoft.DotNet.SDK.6" },
+   	@{name = "Microsoft.VisualStudioCode" }
+);
+Foreach ($app in $apps) {
+    $listApp = winget list --exact -q $app.names
+    if (![String]::Join("", $listApp).Contains($app.name)) {
+        Write-host "Installing: " $app.name
+        winget install -e -h --accept-source-agreements --accept-package-agreements --id $app.name 
```

Comment:
I don't think Microsoft can/should automatically accept binding agreements on behalf of the user. We'll have to remove those flags or somehow show all agreements upfront (which certainly diminishes the value of this script).

Reply:
They aren't necessary at all since none of these packages are coming from the Microsoft Store "msstore" source in WinGet. Since you have the exact PackageIdentifier values called out, the only argument that could be useful here is the -h to get a truly silent install experience as opposed to "silent with progress" which is the default.

Reply:
It might be worth adding "-s winget" to make sure you are using the community repository.

You could also check whether the "winget" source is available via `winget source list` in the event group policy may have restricted or removed the source.

Reply:
That's a good point, Rafael, and thank you for the insights, Demitrius. Because all of my apps are coming fromthe community repository and not MSStore, I got rid of all the agreement acceptance tags. I also got rid of the other "useless" tags. Changes should be reflected in my latest push.